packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -835,16 +836,16 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
       useQuerySubscription,
       useLazyQuerySubscription,
       useLazyQuery(options) {
-        const [trigger, arg] = useLazyQuerySubscription(options)
+        const [trigger, arg, unsubscribe] = useLazyQuerySubscription(options)
         const queryStateResults = useQueryState(arg, {
           ...options,
           skip: arg === UNINITIALIZED_VALUE,
         })
 
         const info = useMemo(() => ({ lastArg: arg }), [arg])
         return useMemo(
-          () => [trigger, queryStateResults, info],
-          [trigger, queryStateResults, info]
+          () => [trigger, queryStateResults, info, unsubscribe],
+          [trigger, queryStateResults, info, unsubscribe]
```

Comment:
I would suggest to make `unsubscribe` part of the `info` object and maybe give it a better name. Open to suggestions.

Reply:
That's some good advice, I missed the fact that `info` was already an object for some extra things so I can touch it without breaking anything 😄 not sure if `extras` is the best possible name though, waiting for further feedback!